skimage/restoration/inpaint.py

```diff
@@ -73,6 +62,30 @@ def _inpaint_biharmonic_single_channel(mask, out, limits):
     return out
 
 
+def _modify_matrices_by_point(mask_pt_idx, mask_pt_n, out_shape, mask, matrix_unknown, matrix_known):
+    # Get bounded neighborhood of selected radius
+    b_lo, b_hi = _get_neighborhood(mask_pt_idx, 2, out_shape)
+
+    # Create biharmonic coefficients ndarray
+    neigh_coef = np.zeros(b_hi - b_lo)
+    neigh_coef[tuple(mask_pt_idx - b_lo)] = 1
+    neigh_coef = laplace(laplace(neigh_coef))
```

Comment:
I think the call to laplace is the only one that is `parallelizable`. The others are all `python` and thus will get held up in the GIL.

Reply:
it seems that each masked point has some variables that are unique to it.
These contain b_lo, b_hi, neight_cief, and the variables it_inner, coefs, tmp_pt_idx and tmp_pt_is.
All of these varibales aren't shared between the processes and thus computing them for each point can be parralilzed in my opinion, as the GIL will not intervene here. Do you agree with me on this point?

I do agree that the GIL might intervene with the modifications of the matrix_unknown and matrix_known variables.
So regarding this, I can suggest the following:
Each process will produce a list of changes that should be made to these matrices, and the main thread will make all of them.
This way the computation of which modification should be made will be done in parallel, and the modification themselves will be done serially. What do you say?